pkg/client/presentproof/client.go

```diff
@@ -0,0 +1,147 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package presentproof
+
+import (
+	"errors"
+
+	"github.com/hyperledger/aries-framework-go/pkg/didcomm/common/service"
+	"github.com/hyperledger/aries-framework-go/pkg/didcomm/protocol/presentproof"
+)
+
+var (
+	errEmptyRequestPresentation = errors.New("request presentation message is empty")
+	errEmptyProposePresentation = errors.New("propose presentation message is empty")
+)
+
+// Provider contains dependencies for the protocol and is typically created by using aries.Context()
+type Provider interface {
+	Service(id string) (interface{}, error)
+}
+
+// ProtocolService defines the presentproof service.
+type ProtocolService interface {
```

Comment:
nitpick: this doesn't need to be exported

Reply:
I am using it for mock generation e.g the usage `mocks.NewMockProtocolService(ctrl)`